scripts/stripe/fix-20240926-recorded-duplicated-orders.ts

```diff
@@ -0,0 +1,153 @@
+import '../../server/env';
+
+import { QueryTypes, Sequelize } from 'sequelize';
+
+import { Order, sequelize as s, Transaction } from '../../server/models';
+
+const sequelize: Sequelize = s;
+
+const AFFECTED_ORDERS_QUERY = `
+    WITH count_payment_intents AS (
+        SELECT o."data"#>>'{previousPaymentIntents,0,id}' AS payment_intent_id, count(1) AS count
+        FROM "Orders" o
+        INNER JOIN "PaymentMethods" pm ON pm.id = o."PaymentMethodId"
+        WHERE TRUE
+        AND pm."service" = 'stripe'
+        AND o.status = 'PAID'
```

Comment:
I don't think filtering on "PAID" orders is enough, I got some entries that had different statuses but were still impacted by the bug with this query shared in https://opencollective.slack.com/archives/C06E23P98DD/p1727440954964609?thread_ts=1727439698.837259&cid=C06E23P98DD:

WITH impacted_orders AS (
  select DISTINCT o1.id, o2.id, o1."FromCollectiveId", o1."updatedAt" as up1, o2."updatedAt" up2
  from "Orders" o1
  INNER JOIN "PaymentMethods" pm ON o1."PaymentMethodId" = pm.id
  INNER JOIN "Orders" o2 ON o1."PaymentMethodId" = o2."PaymentMethodId" AND o1.id < o2.id AND o1.data #>> '{previousPaymentIntents,0,id}' = o2.data #>> '{previousPaymentIntents,0,id}'
  WHERE o1.data #>> '{previousPaymentIntents,0,id}' IS NOT NULL
  AND pm."service" = 'stripe'
  AND (o1."updatedAt" >= '2024-09-25' OR o2."updatedAt" >= '2024-09-25')
) SELECT *
FROM impacted_orders

Reply:
I will add a separate query for Orders that were affected but not marked as PAID because the payment intent field is different in that case.

Reply:
Better query to get them all:

```
WITH impacted_orders AS (
  select DISTINCT o1.id, o2.id, o1."FromCollectiveId", o1."updatedAt" as up1, o2."updatedAt" up2, c.slug
  from "Orders" o1
  INNER JOIN "PaymentMethods" pm ON o1."PaymentMethodId" = pm.id
  INNER JOIN "Orders" o2 ON o1."FromCollectiveId" = o2."FromCollectiveId" AND o1.id < o2.id AND COALESCE(o1.data #>> '{paymentIntent,id}', o1.data #>> '{previousPaymentIntents,0,id}') = COALESCE(o2.data #>> '{paymentIntent,id}', o2.data #>> '{previousPaymentIntents,0,id}')
  INNER JOIN "Collectives" c ON c.id = o1."FromCollectiveId"
  WHERE COALESCE(o1.data #>> '{previousPaymentIntents,0,id}', o1.data #>> '{paymentIntent,id}') IS NOT NULL
  AND pm."service" = 'stripe'
  AND o1."deletedAt" IS NULL
  AND o2."deletedAt" IS NULL
  AND (o1."updatedAt" >= '2024-09-25' OR o2."updatedAt" >= '2024-09-25')
) SELECT *
FROM impacted_orders
```